hal/src/nRF52840/ble_hal.cpp

```diff
@@ -1087,7 +1087,7 @@ int BleObject::Broadcaster::setAdvertisingData(const uint8_t* buf, size_t len) {
     // It is invalid to provide the same data buffers while advertising.
     CHECK(suspend());
     if (buf != nullptr) {
-        len = std::min(len, (size_t)BLE_MAX_ADV_DATA_LEN);
+        len = std::min(len, (size_t)BLE_MAX_ADV_DATA_LEN_EXT);
```

Comment:
We should use `BLE_MAX_ADV_DATA_LEN_EXT_CONNECTABLE` for the advertising data buffer.

Reply:
Same issue as above, I think this is an issue if customer application is non-connectable.